src/transport/raw/tests/TestPeerAddress.cpp

```diff
@@ -39,47 +38,47 @@ using chip::Transport::PeerAddress;
 /**
  *  Test correct identification of IPv6 multicast addresses.
  */
-void TestPeerAddressMulticast(nlTestSuite * inSuite, void * inContext)
+TEST(TestPeerAddress, TestPeerAddressMulticast)
 {
     constexpr chip::FabricId fabric = 0xa1a2a4a8b1b2b4b8;
     constexpr chip::GroupId group   = 0xe10f;
     PeerAddress addr                = PeerAddress::Multicast(fabric, group);
-    NL_TEST_ASSERT(inSuite, chip::Transport::Type::kUdp == addr.GetTransportType());
-    NL_TEST_ASSERT(inSuite, addr.IsMulticast());
+    EXPECT_EQ(chip::Transport::Type::kUdp, addr.GetTransportType());
+    EXPECT_TRUE(addr.IsMulticast());
 
     const Inet::IPAddress & ip = addr.GetIPAddress();
-    NL_TEST_ASSERT(inSuite, ip.IsIPv6Multicast());
-    NL_TEST_ASSERT(inSuite, chip::Inet::IPAddressType::kIPv6 == ip.Type());
+    EXPECT_TRUE(ip.IsIPv6Multicast());
+    EXPECT_EQ(chip::Inet::IPAddressType::kIPv6, ip.Type());
 
     constexpr uint8_t expected[NL_INET_IPV6_ADDR_LEN_IN_BYTES] = { 0xff, 0x35, 0x00, 0x40, 0xfd, 0xa1, 0xa2, 0xa4,
                                                                    0xa8, 0xb1, 0xb2, 0xb4, 0xb8, 0x00, 0xe1, 0x0f };
     uint8_t result[NL_INET_IPV6_ADDR_LEN_IN_BYTES];
     uint8_t * p = result;
     ip.WriteAddress(p);
-    NL_TEST_ASSERT(inSuite, !memcmp(expected, result, NL_INET_IPV6_ADDR_LEN_IN_BYTES));
+    EXPECT_EQ(0, memcmp(expected, result, NL_INET_IPV6_ADDR_LEN_IN_BYTES));
 }
 
-void TestToString(nlTestSuite * inSuite, void * inContext)
+TEST(TestPeerAddress, TestToString)
 {
     char buff[PeerAddress::kMaxToStringSize];
     IPAddress ip;
     {
         IPAddress::FromString("::1", ip);
         PeerAddress::UDP(ip, 1122).ToString(buff);
 
-        NL_TEST_ASSERT(inSuite, !strcmp(buff, "UDP:[::1]:1122"));
+        EXPECT_EQ(0, strcmp(buff, "UDP:[::1]:1122"));
```

Comment:
Use string comparison macro in all cases where `strcmp` is used.

Reply:
Changed.